feat: add fortune telling app (#62167)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-fortune-teller-app/68ed42285f674a9475629831.md

---
id: 68ed42285f674a9475629831
title: Step 17
challengeType: 0
dashedName: step-17
---

# --description--

> Inside the class add another private `elements` as object and inside it add `singleCardBtn`  `singleCard` `multipleCardsBtn` each should have `HTMLElement`.

Reviewer: ```suggestion
Inside the class add another private `elements` as an object and inside it add `singleCardBtn`  `singleCard` `multipleCardsBtn` each should have `HTMLElement`.
```

> 

Reviewer: ```suggestion
```

# --hints--

> You should have declared method named `elements` with `private` method, inside the class.

Reviewer: ```suggestion
You should declare a property named `elements` with the `private` modifier inside the class.
```

```js
assert.match(code, /class\s+\w+\s*{[\s\S]*?private\s+elements\s*:\s*{[\s\S]*?}/);
```

> Inside elements method you should have `singleCardBtn` property should be assigned to `HTMLElement`.

Reviewer: ```suggestion
Inside the `elements` property, the `singleCardBtn` property should be assigned the type `HTMLElement`.
```

```js
assert.match(code, /class\s+\w+\s*{[\s\S]*?private\s+elements\s*:\s*{[\s\S]*?singleCardBtn\s*:\s*HTMLElement\s*;[\s\S]*?}/
);
```

Inside elements method you should have `multipleCardsBtn` property should be assigned to `HTMLElement`.

```js
assert.match(code, /class\s+\w+\s*{[\s\S]*?private\s+elements\s*:\s*{[\s\S]*?multipleCardsBtn\s*:\s*HTMLElement\s*;[\s\S]*?}/);
```

> Inside elements method you should have `singleCard` property assigned should be to `HTMLElement`.

Reviewer: ```suggestion
Inside the `elements` property, the `singleCard` property should be assigned the type `HTMLElement`.
```
